### ISSUE TITLE
Custom keyboard Accelerator abbreviations for all menu items

### ISSUE THREAD

**Issue by eugenesvk** [gildas-lormeau/SingleFile#1420]
**Is your feature request related to a problem? Please describe.**
Currently it's a bit inefficient to pick a specific profile/option from the context menu since you have to look at it and select via a mouse

**Describe the solution you'd like**
Allow custom accelerators so you could open and execute a profile via a sequence of , for example, <kbd>Ctrl</kbd><kbd>A</kbd>,<kbd>G</kbd><kbd>1</kbd> (activate extension, open a `G`roup, select `1Profile`)

Currently the first shortcut doesn't open the extension menu, but launches a page saving operation instead, but that's another feature request https://github.com/gildas-lormeau/SingleFile/issues/1419

**Describe alternatives you've considered (optional)**
Using cursor keys :)

**Additional context (optional)**
N/a

**Comment by gildas-lormeau** [comment]
As discussed in entry #1419, there is no technical way of associating a menu entry with a shortcut. However, I need to see if it would be possible to associate one (or more) for each profile.

**Comment by eugenesvk** [comment]
Well, the mouse context menu subitems can have accelerators, just checked by editing 
```
	"menuSaveWithProfile": {
		"message": "&U Save page with profile",
		"description": "Menu entry: 'Save with profile'"
	},
```

And this works

Though I couldn't quickly find the part that exposes the menu items to the extention button in the toolbar to edit that text and see whether `&A` accelerators are supported

That's already progress as that makes the selection within the menu easier

But then

- is this user-configurable or only editable via the `messages.json`
- is there a way to add an accelerator to the extension item itself? So that  (at least to the page context menu) it's `&SingleFile`
  - I've found that editing the name in the `manifest.json` works!!! Which is great, I'll do that locally, and since you can use external tool to open this context menu via a shortcut I can achieve the full chording experience!!!
-  again, is the ↑ user configurable or only editable via the manifest?

(and I can add `&profile_name` to even have accelerators working with dynamic profile selection menu!)

**Comment by gildas-lormeau** [comment]
Thank you very much for the info, I was not aware of this feature. I'll do some tests and keep you updated.

**Comment by gildas-lormeau** [comment]
I guess I'm missing something here, I tested your `messages.json` file. However, this does not seem to work as expected. On my machine, if I type for example `&` and `d` when the first item is selected in the SingleFile menu then it does not select the "Save selection" entry. I did the test in Firefox and Chrome.

**Comment by eugenesvk** [comment]
Yeah, macOS doesn't support the `&` convention by default (though interestingly enough, the `&` sign isn't shown, so there is some awareness? ).

I don't know how whether there is a workaround/alternative for this (and don't understand why browsers don't support this)

**Comment by gildas-lormeau** [comment]
Two things have landed since you opened this.

Since 1.23.0 each profile can own a keyboard shortcut. In the profile settings there is a `keyboard shortcut command` option where you pick `Custom shortcut #1` to `#10`, and you assign the actual keys in the browser shortcuts page. Pressing it saves with that profile directly. No menu, and one shortcut instead of a sequence. This is also what you asked for in #1419 when you wanted more commands exposed, I just never came back to tell you it was done.

Since 1.24.0 there is a menu editor in Options > Menus. Every entry takes a custom label, and the label reaches the menus API unchanged, so `&D Save selection` sets the access key. That is the accelerator half of this request, per entry and in any locale. It is also why I closed #1425.

What is still not possible is opening the toolbar menu itself from the keyboard, for the reason in #1419. There is no extension API to open a context menu or to bind a shortcut to its entries.

So it depends on which one you actually want. If you want keyboard access to a profile, the custom shortcuts already do it and I can close this. If you want the sequence itself, navigating a menu with letters, the only way would be to replace the toolbar context menu with a popup window. That changes what clicking the button does and means a second menu to maintain, so I would rather know it is really needed before going there.

Which of the two is it?

**Comment by gildas-lormeau** [comment]
To be clear about the second option: a popup is not planned for now.

The layout and the actions themselves would be shared with the existing menus, so it would not be a second menu to define. But it would mean rebuilding in HTML what the native menu gives for free: submenus, checkmarks, keyboard navigation, dismissal, focus and positioning. That is a fair amount of work and I would rather avoid it if I can.

I am keeping the idea rather than ruling it out. If the per-profile shortcuts cover what you need, say so and I will close this.

**Comment by eugenesvk** [comment]
Thank you for adding the extra helpers, I'll need to use them for a while to see whether the imperfections are bothersome, especially in macs without & accelerators

(yeah, rebuilding existing native solutions sounds like an overkill)
